### PR TITLE
Add tropical survival games

### PR DESCRIPTION
**Pull request by yitzy299**
List of biomes:
- Savanna
- Shattered Savanna
- Jungle Hills
- Jungle
- Desert
- Birch forest

### CODE REVIEW

File: src/main/java/supercoder79/survivalgames/game/map/biome/JungleGen.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	}
+
+	@Override
+	public MapGen tree(int x, int z, Random random) {
+		return JungleTreeGen.INSTANCE;
+	}
+
+	@Override
+	public double modifyTreeChance(double original) {
+		return 512;
```
Comment (SuperCoder7979) [inline src/main/java/supercoder79/survivalgames/game/map/biome/JungleGen.java:60]: Higher numbers means fewer trees, is that intentional?

File: src/main/java/supercoder79/survivalgames/game/map/biome/JungleHillsGen.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		return 4.5;
+	}
+
+	@Override
+	public MapGen tree(int x, int z, Random random) {
+		return JungleTreeGen.INSTANCE;
+	}
+
+	@Override
+	public double modifyTreeChance(double original) {
```
Comment (SuperCoder7979) [inline src/main/java/supercoder79/survivalgames/game/map/biome/JungleHillsGen.java:70]: Same here

File: src/main/java/supercoder79/survivalgames/game/map/biome/MesaPlateauGen.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	public double detailFactor() {
+		return 4.5;
+	}
+
+	@Override
+	public MapGen tree(int x, int z, Random random) {
+        if(random.nextInt(2) == 0) {
+            return AcaciaTreeGen.INSTANCE;
+        }
+		return ShrubGen.INSTANCE;
```
Comment (SuperCoder7979) [inline src/main/java/supercoder79/survivalgames/game/map/biome/MesaPlateauGen.java:69]: indentation

File: src/main/java/supercoder79/survivalgames/game/map/biome/MesaPlateauGen.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	@Override
+	public double modifyTreeChance(double original) {
+		return original * 0;
```
Comment (SuperCoder7979) [inline src/main/java/supercoder79/survivalgames/game/map/biome/MesaPlateauGen.java:74]: Just return 0 here

File: src/main/java/supercoder79/survivalgames/game/map/biome/JungleGen.java (hunk 1 above)
Comment (yitzy299) [reply src/main/java/supercoder79/survivalgames/game/map/biome/JungleGen.java:60]: Whoops, let me fix

File: src/main/java/supercoder79/survivalgames/game/map/biome/BadlandsGen.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import net.minecraft.block.Blocks;
+import net.minecraft.util.registry.RegistryKey;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.BiomeKeys;
+import kdotjpg.opensimplex.OpenSimplexNoise;
+
+public class BadlandsGen implements BiomeGen {
+    public static final PlainsGen INSTANCE = new PlainsGen();
+    public static final OpenSimplexNoise TERRACOTTA_NOISE = new OpenSimplexNoise(12);
+    public static final OpenSimplexNoise RED_TERRACOTTA_NOISE = new OpenSimplexNoise(12);
```
[the remark this answers is not in the document]
Comment (yitzy299) [reply src/main/java/supercoder79/survivalgames/game/map/biome/BadlandsGen.java:18]: And spotted another issue as well

**Review by SuperCoder7979 — APPROVED** [review]

**Comment by SuperCoder7979** [comment]
Looks good to me. Thanks!

**Pull request merged**